parsers/test/test_PA.py

```diff
@@ -15,30 +17,94 @@ def setUp(self):
         self.session = requests.Session()
         self.session.mount("https://", self.adapter)
 
-    @freeze_time("2021-12-30 09:58:40", tz_offset=-5)
-    def test_fetch_production(self):
+    def test_fetch_production_live(self):
+        for frozen_time, mock_json_response in [
+            ("2021-12-30 09:58:40", "production_live_20211230.json"),
+            ("2024-04-03 02:43:25", "production_live.json"),
+        ]:
```

Comment:
note: testing against a new recent snapshot but also the historical snapshot to make sure we didn't break anything / code is fully backwards compatible (e.g. can handle legacy fossil fuel plants that may no longer exist)

Reply:
Amazing!! I'll definitely try that in a follow up PR (just to not stack too many changes here)